sklearn/utils/_indexing.py

```diff
@@ -131,7 +133,17 @@ def _pyarrow_indexing(X, key, key_dtype, axis):
         key = np.asarray(key)
 
     if key_dtype == "bool":
+        # TODO: remove version checking and following if-branch when pyarrow==17.0.0 is
+        # the minimal version, see pyarrow issue
+        # https://github.com/apache/arrow/issues/42013 for more info
+        if PYARROW_VERSION_BELOW_17:
+            import pyarrow
+
+            if not isinstance(key, pyarrow.BooleanArray):
+                key = pyarrow.array(key, type=pyarrow.bool_())
```

Comment:
```suggestion
            try:
                pa = sys.modules["pyarrow"]
             except KeyError:  # pragma: no cover
                raise ImportError("pyarrow must be installed to use pyarrow data objects")

            if not isinstance(key, pa.BooleanArray):
                key = pa.array(key, type=pyarrow.bool_())
```

Question for the 2. reviewer: Does it actually make a big enough difference (compared to `import pyarrow`)?

Reply:
In that case you can just do `pa = sys.modules["pyarrow"]` without the try except.